acra-censor/handlers/blacklist_handler.go

```diff
@@ -342,3 +347,37 @@ func handleValuePattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
 	}
 	return matchDetected
 }
+
+//handle SELECT * FROM table %%WHERE%% pattern
+func handleStarPattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
+	patternStarDetected := false
+	patternWhereDetected := false
+	queryWhereDetected := false
+	var patternTables sqlparser.TableExprs
+	var queryTables sqlparser.TableExprs
+	for _, patternNode := range patternNodes {
+		if _, ok := patternNode.(*sqlparser.StarExpr); ok {
+			patternStarDetected = true
+		}
+		if patternWhereNode, ok := patternNode.(*sqlparser.Where); ok && patternWhereNode != nil && strings.EqualFold(sqlparser.String(patternWhereNode.Expr), WhereConfigPlaceholderReplacerPart2) {
+			patternWhereDetected = true
+		}
+		if patternSelectStatement, ok := patternNode.(*sqlparser.Select); ok && patternSelectStatement != nil {
+			patternTables = patternSelectStatement.From
+		}
+	}
+	for _, queryNode := range queryNodes {
+		if queryWhereNode, ok := queryNode.(*sqlparser.Where); ok && queryWhereNode != nil {
+			queryWhereDetected = true
+		}
+		if querySelectStatement, ok := queryNode.(*sqlparser.Select); ok && querySelectStatement != nil {
+			queryTables = querySelectStatement.From
+		}
+	}
+	if patternStarDetected {
```

Comment:
can we move this check after walking through patternNodes and write `if !patternStarDetected{ return false}` to avoid processing second loop?